File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int answer = 0;
    
    for(int i = 0; i < s.size(); ++i) {
        if (s[i] == '(') answer += 1;
        else {
            if (answer > 0) answer -= 1;
            else return false;
        }
    }
    if (answer > 0) return false;
    return true;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int answer = 0;

        for (int i = 0; i < s.length(); ++i) {
            if (s.charAt(i) == '(') {
                answer += 1; 
            } else {
                if (answer > 0) {
                    answer -= 1;
                } else {
                    return false;
                }
            }
        }
        return answer == 0;
    }
}